msal-go/advanced/managed-identity.md

```diff
@@ -0,0 +1,148 @@
+
+## Creating Azure Resources
+
+You can create the resources needed for running the sample manually via the [Azure Portal](https://portal.azure.com/#home), or for a quick and
+concise breakdown of how to do it using Azure CLI or Azure Powershell, follow the instructions [here](/azure/developer/go/azure-sdk-authentication-managed-identity?tabs=azure-cli)
+
+## Quick start
+
+To quickly get started and see Azure Managed Identity in action, you can use one of the samples the team built for this purpose:
+
+> [Use Managed Identity sample](https://github.com/Azure-Samples/msal-managed-identity/tree/main/src/go)
+
+## Running Locally
+
+### Step 1:  Clone or download this repository
+
+From your shell or command line:
+
+```Shell
+git clone https://github.com/AzureAD/microsoft-authentication-library-for-go.git
+```
+
+or download and extract the repository `.ZIP` file.
+
+The GO sample is located in the [`/apps/tests/devapps/managedidentity`](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/tests/devapps/managedidentity/managedidentity_sample.go) folder.
+
+### Step 2:  Modify the Key Vault URI and Secret name values in the code
+
+Following are the changes you need to make:
+
+- In the [`managedidentity_sample.go`](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/andyohart/managed-identity/apps/tests/devapps/managedidentity/managedidentity_sample.go) file under the getSecretFromAzureVault method modify the following values,
+
+    ```go
+    keyVaultUri := "your-key-vault-uri"
+    secretName := "your-secret-name"
+    ```
+
+- Change these to match your key vault uri and secret name. These can be found in the following locations:
+
+1. **Key Vault URI** - In your Azure home page, go to your key vault, on the Overview page our key vault URI can be found under **'Essentials'**
+1. **Secret Name** - On the Key Vault Overview page, go to the Panel on the left and expand the **'Objects'** dropdown  
+Click into **'Secrets'**  
+Click into the secret you want to use  
+Click ont the version you would like to use  
+Copy the part after the key vault URI and use that as your secret name  
+
+### Step 3:  Save and Run the Sample
+
+1. Save the file  
+1. In your terminal navigate to the `/apps/tests/devapps/managedidentity` directory  
+1. Run the sample with the command
+    ```
+        go run .
+    ```  
+1. The sample you want to run is the one corresponding to option **'9'**, so when prompted enter that and it should run and show you the result
```

Comment:
I am not entirely sure why we call out the specific option here, rather than what it should be about (managed identity).